### ISSUE TITLE
matTextareaAutosize  and matAutosizeMinRows don't work in textarea

### ISSUE THREAD

**Issue by drenda** [creativetimofficial/material-dashboard-pro-angular#107]
I've a problem with TextArea. I'm using this code

`<mat-form-field> <textarea matInput placeholder="{{ 'note.labels.note' | translate }}" [(ngModel)]="notes" matTextareaAutosize matAutosizeMinRows="3" matAutosizeMaxRows = "5" name="notes" #note></textarea> </mat-form-field> `

where I'm using `matAutosizeMinRows` & `matAutosizeMaxRows` properties (https://material.angular.io/components/input/api).
I thought was my fault and I opened a ticket on stackoverflow (https://stackoverflow.com/questions/46642234/material-textarea-dont-fit-minrows?noredirect=1#comment80237144_46642234) but my code is right and putting that on a pluncker it works. Instead inside your theme that properties don't work. 

Could you check please? Thanks

**Comment by chelaruc** [comment]
@drenda Can you give me the plunker?

**Comment by drenda** [comment]
Sure, here it is: https://stackblitz.com/edit/ng-mat-beta-11-hcvaxg

**Comment by chelaruc** [comment]
@drenda I have tested with this code:
```
<mat-form-field>
        <textarea placeholder="aici" [(ngModel)]="notes" matInput matAutosizeMinRows="6" matAutosizeMaxRows="10" name="notes" #note matTextareaAutosize></textarea> 
</mat-form-field>
````
and it worked for me. What problem do you have?

Best,
Ciprian

**Comment by drenda** [comment]
You are right, your code work fine in first level views. It doesn't work if it is inside module imported in the main view. Do you know if is there something to keep in mind to make it work? Thanks

**Comment by chelaruc** [comment]
@drenda Can you give an archive to test it?

**Comment by drenda** [comment]
[Here](https://drive.google.com/open?id=0B_she6Q8vsHrQTlpaWF6aDgtZW8) you can find an example of the problem. I found the problem is when the textarea is inside a tab. Check the example. Thanks

**Comment by chelaruc** [comment]
Hi, @drenda! Sorry for the late response! You are right, it does not work inside a tab. But I think will work with [Material Tabs](https://material.angular.io/components/tabs/examples). 

Best,
Ciprian

**Comment by drenda** [comment]
Hi, are you planning to convert your tabs to Material tabs? Thanks

**Comment by jpduckwo** [comment]
I think it also doesn't work inside mat-expansion-panel... when it expands all my textareas are only 1 row

**Comment by AkashShivanand** [comment]
I am using it inside mat-tab but, it's not working.

**Comment by lippomano** [comment]
@chelaruc any news?

**Comment by chelaruc** [comment]
I still do not have a solution for this

**Comment by jpduckwo** [comment]
I've been looking into this and I think the core reason behind it not working is that the text area size is calculated before the tab has completed animating in. If someone wants to dig further I would suggest hooking into animationDone output on mat-tab-group and kicking off a change detection using ChangeDetectorRef.detectChanges() in the component that contains the mat-tab-group.

see: https://material.angular.io/components/tabs/api#MatTabGroup

**Comment by decision-ca** [comment]
I found a temporary solution.
I commented the property `height: auto` of the file **/assets/scss/core/_forms.scss**

```
textarea{
    // height: auto !important;
    resize: none;
    line-height: $bmd-line-height !important;
}
```
And the html: 
```
<mat-form-field>
   <textarea matInput cdkTextareaAutosize 
         cdkAutosizeMinRows="2" cdkAutosizeMaxRows="5" 
         placeholder="Observación" maxlength="512">
   </textarea>
</mat-form-field>
```

I hope it helps you solve your mistakes

**Comment by rushikesh07** [comment]
use cdkTextareaAutosize instade of matTextareaAutosize

**Comment by stormlightstudio** [comment]
I was using `cdkTextareaAutosize` and `matAutosizeMinRows` which was not working.

Switching to `cdkTextareaAutosize` and `cdkAutosizeMinRows` works as expected for me.
